### ISSUE TITLE
Immediate crash with .dng from Samsung S21 Ultra

### ISSUE THREAD

**Issue by wilecoyote2015** [darktable-org/darktable#11847]
**Describe the bug/issue**
In lighttable, zooming in with the image selected leads to a crash.
Also, double-click to open the image in the darkroom results in a crash.

**To Reproduce**
_Please provide detailed steps to reproduce the behaviour, for example:_
1. Import the attached DNG file
[crash_dng.zip](https://github.com/darktable-org/darktable/files/8731714/crash_dng.zip)
3. Zoom in in lighttable or open the image in darkroom

**Platform**
_Please fill as much information as possible in the list given below. Please state "unknown" where you do not know the answer and remove any sections that are not applicable _

* darktable version : 3.9.0+1539~g5ff5ac0349
* OS                : arch linux
* Memory            : 40GB
* Graphics card     : Intel Xe iGPU
* OpenCL installed  :  yes
* OpenCL activated  : tried with and without
* Xorg              : wayland
* Desktop           : sway
* gcc               : 11.2.0
* cflags            :  See https://aur.archlinux.org/packages/darktable-git

**Additional context**
Log: 
[crash_dng.txt](https://github.com/darktable-org/darktable/files/8731711/crash_dng.txt)

 - Can you reproduce with a RAW or Jpeg or both? Only dng
 - Are the steps above reproducible with a fresh edit (i.e. after discarding history)?  yes
 - Is the issue still present using an empty/new config-dir (e.g. start darktable with --configdir "/tmp")? yes

**Comment by todd-prior** [comment]
Seems to be working okay in Win 10...

![image](https://user-images.githubusercontent.com/44409419/169385119-71a98135-450a-442c-b9f8-48ede9ecf308.png)

**Comment by pehar1** [comment]
3.9.0+1551~g79adc8034, Ubuntu 20.04, OpenCL enabled
Image loads without any problems into lighttable, can be edited in darkroom.

**Comment by kmilos** [comment]
Sounds a bit like the macOS problem w/ DNGs mentioned here (those also work ok on Windows btw):

https://discuss.pixls.us/t/current-osx-build/13213/445 (and below)

Someone who can reproduce this on Linux/macOS will just have to build and run in debug to get us more info...

What I don't get is, why the GraphicsMagick delegate is involved here at all? (i.e. the DNG format and thumbnail extraction should be supported without GraphicsMagick help IMHO...)

Just as a data point: the [Windows builds use GraphicsMagick 1.3.38 w/ Q16](https://packages.msys2.org/package/mingw-w64-x86_64-graphicsmagick?repo=mingw64)

**Comment by kmilos** [comment]
Ah, it falls through to the backup GraphicsMagick thumbnail extraction method because the regular thumbnail extraction method via exiv2 fails:

a) thumbnail is not a JPEG (seems to be uncompressed?)
b) thumbnail has corrupt tags (`Warning: Directory SubImage1, entry 0x0111: Strip 0 is outside of the data area; ignored.`)

So Samsung is not really producing 100% kosher files here (someone should let them know), but that doesn't mean it should crash dt indeed (we could just return no thumbnail)... And why it crashes only Linux/macOS and not Windows...?

**Comment by kmilos** [comment]
For the record, this thumbnail written by Samsung really doesn't match up in more than one way:

```
Exif.SubImage1.NewSubfileType                Long        1  Thumbnail/Preview image
Exif.SubImage1.ImageWidth                    Long        1  504
Exif.SubImage1.ImageLength                   Long        1  378
Exif.SubImage1.BitsPerSample                 Short       1  12
Exif.SubImage1.Compression                   Short       1  Uncompressed
Exif.SubImage1.PhotometricInterpretation     Short       1  YCbCr
Exif.SubImage1.StripOffsets                  Long        1  24025952
Exif.SubImage1.SamplesPerPixel               Short       1  3
Exif.SubImage1.RowsPerStrip                  Long        1  378
Exif.SubImage1.StripByteCounts               Long        1  571536
Exif.SubImage1.PlanarConfiguration           Short       1  Chunky
```

a) StripOffsets 24025952 + StripByteCounts 571536 > actual file size 24311720 (hence the exiv2 warning above)

b) SamplesPerPixel is 3, but there is only a single BitsPerSample

c) No sub-sampling is specified while the number of bytes seem to indicate 4:2:0 (571536 = 504*378*1.5*2) - this might be ok since that is the default

d) Exactly half of the bytes are actually written (285768), so what's in them? Just the luma channel? But then it has to be packed (504*378*12/8), which no longer means "Uncompressed"

In short, this thumbnail is a mess, but should nonetheless be safely skipped by dt.
